src/views/modal_handler.rs

```diff
@@ -59,11 +64,19 @@ pub(crate) const MODAL_CONTAINER_STYLE: ContainerStyle = ContainerStyle {
     border_radius: 2.0,
     border_width: 2.0,
 };
+const HYPERLINK_BUTTON_STYLE: ButtonStyle = ButtonStyle {
+    bg_color: Color::TRANSPARENT,
+    text_color: Color::from_rgba(0.0, 0.3, 0.8, 1.0),
+    border_radius: 2.0,
+    hovered_bg_color: Color::TRANSPARENT,
+    hovered_text_color: Color::from_rgba(0.0, 0.0, 0.6, 1.0),
+};
 
 impl DisplayModal {
     pub fn new() -> Self {
         Self {
             show_modal: false,
+            is_warning: false,
```

Comment:
Mor ebooleans creeping in here.

I suggest you have an enum (`DialogType`) that has variants for each of the dialogs to show (HardwareDetails, Version, Exit..) (containing information specific to each one if that is applicable), and then have a message ShowDialog(DialogType) that is sent.

If needed, you can store the current dialog being shown in a self.dialog_type : DialogType field (but not sure if needed?).